From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Chris Leech <cleech@redhat.com>
Date: Thu, 2 Jan 2020 14:24:43 -0500
Subject: [PATCH] be2iscsi: remove unsupported device IDs

Message-id: <20180503194032.1461134-1-cleech@redhat.com>
Patchwork-id: 212454
O-Subject: [RHEL8 PATCH e-stor] scsi: be2iscsi: remove unsupported device IDs
Bugzilla: 1574502
RH-Acked-by: Maurizio Lombardi <mlombard@redhat.com>
RH-Acked-by: Tom Coughlan <coughlan@redhat.com>
RH-Acked-by: Ewan Milne <emilne@redhat.com>

Bugzilla: 1574502
Upstream Status: RHEL Specific
Build Info: brew 15975638

BladeEngine 2 generation devices are deprecated in RHEL 7
and not supported in RHEL 8

Signed-off-by: Chris Leech <cleech@redhat.com>
Signed-off-by: Herton R. Krzesinski <herton@redhat.com>

be2iscsi: remove BE3 family support

Message-id: <1530890927-5792-1-git-send-email-mlombard@redhat.com>
Patchwork-id: 223576
O-Subject: [RHEL8 e-stor PATCH] be2iscsi: remove BE3 family support
Bugzilla: 1598366
RH-Acked-by: Tomas Henzl <thenzl@redhat.com>
RH-Acked-by: Chris Leech <cleech@redhat.com>

BZ: https://bugzilla.redhat.com/show_bug.cgi?id98366
Brew: https://brewweb.engineering.redhat.com/brew/taskinfo?taskID996775
Upstream: RHEL-only, they tried to mark BE3 as unsupported
          in upstream kernel but the patch has been rejected.
          See: https://marc.info/?l=linux-scsi&m2835991600881&w=2

Broadcom doesn't want to provide support for the BE3 device family
in RHEL8.

This patch removes the BE3's PCI-IDs.

Upstream Status: RHEL only
Signed-off-by: Maurizio Lombardi <mlombard@redhat.com>
Signed-off-by: Herton R. Krzesinski <herton@redhat.com>
---
 drivers/scsi/be2iscsi/be_main.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/scsi/be2iscsi/be_main.c b/drivers/scsi/be2iscsi/be_main.c
index 9b81cfbbc5c5..1ca08297939c 100644
--- a/drivers/scsi/be2iscsi/be_main.c
+++ b/drivers/scsi/be2iscsi/be_main.c
@@ -370,11 +370,13 @@ static int beiscsi_eh_device_reset(struct scsi_cmnd *sc)

 /*------------------- PCI Driver operations and data ----------------- */
 static const struct pci_device_id beiscsi_pci_id_table[] = {
+#ifndef CONFIG_RHEL_DIFFERENCES
 	{ PCI_DEVICE(BE_VENDOR_ID, BE_DEVICE_ID1) },
 	{ PCI_DEVICE(BE_VENDOR_ID, BE_DEVICE_ID2) },
 	{ PCI_DEVICE(BE_VENDOR_ID, OC_DEVICE_ID1) },
 	{ PCI_DEVICE(BE_VENDOR_ID, OC_DEVICE_ID2) },
 	{ PCI_DEVICE(BE_VENDOR_ID, OC_DEVICE_ID3) },
+#endif
 	{ PCI_DEVICE(ELX_VENDOR_ID, OC_SKH_ID1) },
 	{ 0 }
 };
-- 
2.26.2

